src/components/JUPTable/JUPTable.tsx

```diff
@@ -57,92 +49,136 @@ function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+function getComparator(order: Order, orderBy: "toString" | "valueOf"): (a: ITableRow, b: ITableRow) => number {
   return order === "desc" ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
 }
 
 interface IEnhancedTableProps {
-  // onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Data) => void;
   onRequestSort: (event: React.MouseEvent<unknown>, property: string) => void;
   order: Order;
   orderBy: string;
-  headCells: Array<IHeadCellProps>;
+  headCells?: Array<IHeadCellProps>;
 }
 
 const EnhancedTableHead: React.FC<IEnhancedTableProps> = ({ onRequestSort, order, orderBy, headCells }) => {
-  const createSortHandler = (property: keyof Data) => (event: React.MouseEvent<unknown>) => {
-    onRequestSort(event, property);
-  };
+  const createSortHandler = useCallback(
+    (event: React.MouseEvent<unknown>, property: string) => {
+      onRequestSort(event, property);
+    },
+    [onRequestSort]
+  );
+
+  const HeadCellsMemo = useMemo(() => {
+    return headCells?.map((headCell) => (
+      <TableCell key={headCell.id} align={headCell.headAlignment} padding={"normal"} sortDirection={orderBy === headCell.id ? order : false}>
+        <TableSortLabel
+          active={orderBy === headCell.id}
+          direction={orderBy === headCell.id ? order : "asc"}
+          onClick={(e) => createSortHandler(e, headCell.id)}
+        >
+          {headCell.label}
+          {orderBy === headCell.id ? (
+            <Box component="span" sx={visuallyHidden}>
+              {order === "desc" ? "sorted descending" : "sorted ascending"}
+            </Box>
+          ) : null}
+        </TableSortLabel>
+      </TableCell>
+    ));
+  }, [createSortHandler, headCells, order, orderBy]);
 
   return (
     <TableHead>
-      <TableRow>
-        {headCells.map((headCell) => (
-          <TableCell
-            key={headCell.id}
-            align={headCell.numeric ? "right" : "left"}
-            padding={headCell.disablePadding ? "none" : "normal"}
-            sortDirection={orderBy === headCell.id ? order : false}
-          >
-            <TableSortLabel
-              active={orderBy === headCell.id}
-              direction={orderBy === headCell.id ? order : "asc"}
-              onClick={createSortHandler(headCell.id)}
-            >
-              {headCell.label}
-              {orderBy === headCell.id ? (
-                <Box component="span" sx={visuallyHidden}>
-                  {order === "desc" ? "sorted descending" : "sorted ascending"}
-                </Box>
-              ) : null}
-            </TableSortLabel>
-          </TableCell>
-        ))}
-      </TableRow>
+      <TableRow>{HeadCellsMemo}</TableRow>
     </TableHead>
   );
 };
 
 interface IJUPTableProps {
-  headCells: Array<IHeadCellProps>;
-  rows: Array<React.ReactElement>;
   title: string;
-  path: string;
+  headCells?: Array<IHeadCellProps>;
```

Comment:
i'll look into adding some protection code tonight that helps debug / error when the headcells.id has no match in the rows data. this is one of the only 'manually' typed areas we have so far. i didn't know how to get ts to make this happen the other day when doing this code, but i want to circle back to it and figure it out. it can be done separately from this PR so you don't have to wait for me.

Reply:
ok yea that would be excellent because getting that data situated when adding new tables has been painful